lib/bitcoinerlab.js

```diff
@@ -0,0 +1,144 @@
+const secp256k1 = require('@exodus/bitcoinerlab-secp256k1')
+let elliptic
+
+const verbose = false
+function loadLegacy (method, message) {
+  if (verbose) {
+    if (!message) {
+      console.warn(`${method} is not implemented, using legacy approach...`)
+    } else console.warn(`${method} | ${message}`)
+  }
+  if (!elliptic) {
+    elliptic = require('./elliptic')
+  }
+}
+
+function isCompressedPublicKey (array) {
+  return array.length === 33
+}
+
+module.exports = {
+  contextRandomize: function () {
+    return 0
+  },
+  privateKeyVerify: function (privateKey) {
+    return secp256k1.isPrivate(privateKey) ? 0 : 1
+  },
+  privateKeyNegate: function (privateKey) {
+    try {
+      privateKey.set(secp256k1.privateNegate(privateKey))
+    } catch (e) {
+      loadLegacy('privateKeyNegate', e.toString())
+      elliptic.privateKeyNegate(privateKey)
+    }
+    return 0
+  },
+  privateKeyTweakAdd: function (privateKey, tweak) {
+    try {
+      privateKey.set(secp256k1.privateAdd(privateKey, tweak))
```

Comment:
`secp256k1.privateAdd(privateKey)` can return null, please throw on that rather than setting `privateKey`  to null.

There is a difference in behavior here it seems:
* [elliptic seems to wrap on overflow with isub](https://github.com/ExodusMovement/secp256k1-node/blob/master/lib/elliptic.js#L91-L92). Which is interesting, as it doesn't guarantee that our BN is within N 😕 I would expect a umod here like with `privateKeyNegate`
* [bitcoinerlab also wraps on overflow, but seems like it does properly use `mod`](https://github.com/bitcoinerlab/secp256k1/blob/main/index.js#L114-L115)